File: app/src/main/java/com/gmail/fattazzo/formula1world/ergast/json/objects/Time.java
package com.gmail.fattazzo.formula1world.ergast.json.objects;

import com.gmail.fattazzo.formula1world.domain.F1Time;

/**
 * @author fattazzo
 *         <p/>
 *         date: 06/06/17
 */
public class Time {
    private int millis;
    private String time;

    public Time(int millis, String time) {
        this.millis = millis;
        this.time = time;
    }

    F1Time toF1Time() {
        F1Time f1Time = new F1Time();
        f1Time.millis = this.millis;
        f1Time.time = this.time;
        return f1Time;
    }

    @Override
    public String toString() {
        return "Time{" +
                "millis=" + millis +
                ", time='" + time + '\'' +
                '}';
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/ergast/json/objects/Time.kt
/*
 * Project: total-gp-world
 * File: Time.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.ergast.json.objects

import com.gmail.fattazzo.formula1world.domain.F1Time

/**
 * @author fattazzo
 *
 *
 * date: 06/06/17
 */
class Time(private val millis: Int, private val time: String) {

    internal fun toF1Time(): F1Time {
        val f1Time = F1Time()
        f1Time.millis = this.millis
        f1Time.time = this.time
        return f1Time
    }

    override fun toString(): String {
        return "Time{" +
                "millis=" + millis +
                ", time='" + time + '\'' +
                '}'
    }
}
